### PR TITLE
Enable Travis, add analysis_options, update .gitignore

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  unused_import: error
+  unused_local_variable: error
+  dead_code: error
+  ### Useful to uncomment during development to remove the noise of the many
+  ### deprecated APIs.
+  # deprecated_member_use: ignore
+linter:
+  rules:
+    - await_only_futures
+    - implementation_imports
```
Comment (nex3) [inline analysis_options.yaml]: No lints.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+language: dart
+
+dart:
+  - dev
+  - stable
+
+dart_task:
+  - test
+  - dartanalyzer
```
Comment (nex3) [inline .travis.yml]: Only run against the dev analyzer.

Why aren't you using [the `.travis.yml` file from `oss_project_templates`](https://github.com/dart-lang/oss_project_templates/blob/master/dart-github-dart-lang/.travis.yml)?

File: analysis_options.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+analyzer:
+  strong-mode: true
+# Upgrade lints to errors that will break Google3
```
Comment (nex3) [inline analysis_options.yaml]: Why do these break Google3 if they aren't errors externally? They don't affect downstream packages in any way.

File: analysis_options.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+errors:
+  unused_element: error
+  unused_import: error
```
Comment (nex3) [inline analysis_options.yaml]: I'm fairly certain this at least *doesn't* break Google3. I'm pretty sure we've had packages that ended up with unused imports lasting a long time without breaking anything.

File: analysis_options.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  unused_local_variable: error
+  dead_code: error
+  ### Useful to uncomment during development to remove the noise of the many
+  ### deprecated APIs.
+  # deprecated_member_use: ignore
```
Comment (nex3) [inline analysis_options.yaml]: What "many deprecated APIs"? Why would we want to ignore deprecation warnings?

File: pubspec.yaml [hunk 6]
```diff
@@ -1,5 +1,5 @@
 name: term_glyph
-version: 1.0.0
+version: 1.0.1-dev
```
Comment (nex3) [inline pubspec.yaml]: You don't need to bump the pubspec for an infrastructure-only change.

**Comment by kevmoo** [comment]
PTAL

**Pull request merged**